tests/test_signals.py

```diff
@@ -178,3 +181,95 @@ def test_empty_event_data_format_skips_processing(self):
+        """
+        Verify that `XBLOCK_SKILL_VERIFIED` signal is handled as expected.
+        """
+        # setup, create xblock and related skills
+        get_product_skills_mock.return_value = self.skills_emsi_client_response
+        get_block_provider_mock.return_value = DiscoveryXBlockMetadataProvider([self.xblock])
+        XBLOCK_PUBLISHED.send_event(
+            xblock_info=XBlockData(
+                usage_key=self.xblock.key,
+                block_type=self.xblock.content_type,
+            ),
+        )
+
+        self.assertEqual(XBlockSkillData.objects.all().count(), 4)
+        self.assertEqual(sum(XBlockSkillData.objects.all().values_list("verified_count", flat=True)), 0)
+        self.assertEqual(sum(XBlockSkillData.objects.all().values_list("ignored_count", flat=True)), 0)
+        skill_ids = list(XBlockSkillData.objects.all().values_list("skill_id", flat=True))
+
+        # action
+        XBLOCK_SKILL_VERIFIED.send_event(
+            xblock_info=XBlockSkillVerificationData(
+                usage_key=self.xblock.key,
+                verified_skills=skill_ids[:2],
+                ignored_skills=skill_ids[2:],
+            )
+        )
+        # verify
+        self.assertEqual(list(XBlockSkillData.objects.all().values_list("verified_count", flat=True)), [1, 1, 0, 0])
+        self.assertEqual(list(XBlockSkillData.objects.all().values_list("ignored_count", flat=True)), [0, 0, 1, 1])
+
+        # double check
+        XBLOCK_SKILL_VERIFIED.send_event(
+            xblock_info=XBlockSkillVerificationData(
+                usage_key=self.xblock.key,
+                verified_skills=skill_ids[:2],
+                ignored_skills=skill_ids[2:],
+            )
+        )
+        self.assertEqual(list(XBlockSkillData.objects.all().values_list("verified_count", flat=True)), [2, 2, 0, 0])
+        self.assertEqual(list(XBlockSkillData.objects.all().values_list("ignored_count", flat=True)), [0, 0, 2, 2])
+
+    def test_missing_xblock_skills_verification_signals(self):
+        """
+        Verify that `XBLOCK_SKILL_VERIFIED` signal does nothing if xblock does not exists.
+        """
+
+        with LogCapture(level=logging.INFO) as log_capture:
+            XBLOCK_SKILL_VERIFIED.send_event(
+                xblock_info=XBlockSkillVerificationData(
+                    usage_key=self.xblock.key,
+                    verified_skills=[1],
+                    ignored_skills=[],
+                )
+            )
+            messages = [record.msg for record in log_capture.records]
+            self.assertEqual(len(log_capture.records), 6)
+            self.assertEqual(
+                [
+                    f'[TAXONOMY] XBLOCK_SKILL_VERIFIED signal received',
+                    '[TAXONOMY] update_xblock_skills_verification_counts task triggered',
+                    f'[TAXONOMY] XBlock with usage_key: {self.xblock.key} not found!',
```

Comment:
Shouldn't this contain the task completed log as well, if we're returning after logging error message?

Reply:
Yes, it was not checking for it, see next line: `messages[:-3]`. Updated it to check for it, thanks.